platforms/web/lib/conversion.test.ts

```diff
@@ -98,3 +98,49 @@ describe('Plain text <=> markdown', () => {
         expect(convertedPlainText).toBe(expectedPlainText);
     });
 });
+
+describe('Mentions conversion from plain text to rich', () => {
+    it('converts at-room mentions as expected', async () => {
+        const input = '@room';
+        const asComposerHtml = await plainToRich(input);
+        const asMessageHtml = await plainToRich(input, true);
+
+        // for this case, we'd expect the composer html to be an <a> tag, whereas
+        // the message html should be the equivalent plain text
+        expect(asComposerHtml).toBe(
+            '<a data-mention-type="at-room" href="#" contenteditable="false">@room</a>',
+        );
+        expect(asMessageHtml).toBe('@room');
+    });
```

Comment:
nitpick:

In these cases, it feels like there are two things being tested per test. I'd be tempted to split this into two e.g.

```typescript
    const atRoomInput = '@room';
    
    it('converts at-room mentions to message html', async () => {
        const asMessageHtml = await plainToRich(atRoomInput, true);

        expect(asMessageHtml).toBe('@room');
    });
    
    it('converts at-room mentions to composer html', async () => {
        const asComposerHtml = await plainToRich(atRoomInput);

        // for this case, we'd expect the composer html to be an <a> tag, whereas
        // the message html should be the equivalent plain text
        expect(asComposerHtml).toBe(
            '<a data-mention-type="at-room" href="#" contenteditable="false">@room</a>',
        );
    });
```

This way, when something breaks, it's easier to see exactly which input causes the failure.

Reply:
I think that's a fair point, will separate them out